src/lib.rs

```diff
@@ -46,15 +55,35 @@ fn my_cgroup() -> Result<PathBuf> {
     return Ok(PathBuf::from(cgroups[4..].trim()));
 }
 
+// Starting from the base_cg, traverses the cgroup heirarchy and returns the first set/non-empty
+// contents of the given cgroup controller.
+// Returns an error if there is no such file.
+fn read_cg_controller(base_cg: PathBuf, controller_name: &str) -> Result<String> {
+    let mut cur = Some(base_cg.as_path());
+
+    while let Some(cg) = cur {
+        let path: PathBuf = ["/sys/fs/cgroup", &cg.to_string_lossy(), controller_name]
+            .iter()
+            .collect();
+
+        match fs::read_to_string(&path) {
+            Ok(v) if !v.trim().is_empty() => return Ok(v),
+            _ => cur = cg.parent(),
+        }
+    }
+
+    Err(Errno { errno: ENOENT })
+}
+
 fn parse_effective_cpus(raw_cpus: String) -> Result<c_int> {
     let intervals = raw_cpus.trim().split(',');
     let counts = intervals
         .map(|i| -> Result<c_int> {
             let membs: Vec<_> = i.split('-').collect();
             if let [start, end] = &membs[..] {
                 Ok(end.parse::<c_int>()? - start.parse::<c_int>()? + 1)
-            } else if let [single] = &membs[..] {
-                Ok(single.parse::<c_int>()?)
+            } else if let [_single] = &membs[..] {
```

Comment:
Why _?

Reply:
rust was complaining about an unused variable, this was the suggested fix

Reply:
Oh, right. In that case you can just call it `_`, which I think is the idiom for unused.

Reply:
done